src/java/org/apache/cassandra/service/accord/journal/AccordTopologyUpdate.java

```diff
@@ -55,178 +56,181 @@ static AccordTopologyUpdate newTopology(Journal.TopologyUpdate update)
             {
                 size += TypeSizes.sizeofUnsignedVInt(e.getKey());
-                size += RangesForEpochSerializer.instance.serializedSize(e.getValue(), version);
+                size += RangesForEpochSerializer.instance.serializedSize(e.getValue());
             }
 
-            size += TopologySerializers.topology.serializedSize(from.local, version);
-            size += TopologySerializers.topology.serializedSize(from.global, version);
+            size += TopologySerializers.topology.serializedSize(from.local);
+            size += TopologySerializers.topology.serializedSize(from.global);
             return size;
         }
     }
 
-    class Serializer implements IVersionedSerializer<AccordTopologyUpdate>
+    class Serializer implements org.apache.cassandra.io.Serializer<AccordTopologyUpdate>
     {
         public static Serializer instance = new Serializer();
 
-        public void serialize(AccordTopologyUpdate t, DataOutputPlus out, int version) throws IOException
+        @Override
+        public void serialize(AccordTopologyUpdate t, DataOutputPlus out) throws IOException
         {
             out.writeUnsignedVInt(t.epoch());
             out.writeUnsignedVInt32(t.kind().ordinal());
             switch (t.kind())
             {
                 case NewTopology:
-                    TopologyUpdateSerializer.instance.serialize(((NewTopology) t).update, out, version);
+                    TopologyUpdateSerializer.instance.serialize(((NewTopology) t).update, out);
                     break;
                 case Topologies:
                     TopologyImage image = (TopologyImage) t;
 
                     out.writeBoolean(image.update != null);
                     if (image.update != null)
-                        TopologyUpdateSerializer.instance.serialize(image.update, out, version);
+                        TopologyUpdateSerializer.instance.serialize(image.update, out);
                     if (image.syncStatus == null)
                         out.writeByte(Byte.MAX_VALUE);
                     else
                         out.writeByte(image.syncStatus.ordinal());
 
-                    KeySerializers.ranges.serialize(image.closed, out, version);
-                    KeySerializers.ranges.serialize(image.retired, out, version);
+                    KeySerializers.ranges.serialize(image.closed, out);
+                    KeySerializers.ranges.serialize(image.retired, out);
                     break;
                 default:
                     throw new UnhandledEnum(t.kind());
             }
         }
 
-        public AccordTopologyUpdate deserialize(DataInputPlus in, int version) throws IOException
+        @Override
+        public AccordTopologyUpdate deserialize(DataInputPlus in) throws IOException
         {
-            int epoch = in.readUnsignedVInt32();
+            long epoch = in.readUnsignedVInt();
```

Comment:
bug, we read an epoch > int max which then failed in the tests.  We write long, sizeOf is long, yet read int?